scipy/stats/_hypotests.py

```diff
@@ -890,29 +897,42 @@ def barnard_exact(table, alternative="two-sided", pooled=True, num_it=3):
         raise ValueError(msg)
 
     p_value = _binomial_maximisation_of_p_value_with_nuisance_param(
-        total_c1, total_c2, idx, num_it
+        total_col_1, total_col_2, index_arr, n_iter
     )
-    return BarnardExactResult(TX_obs, p_value)
+    return BarnardExactResult(wald_stat_obs, p_value)
 
 
 def _binomial_maximisation_of_p_value_with_nuisance_param(
-    total_c1, total_c2, idx, num_it
+    total_c1, total_c2, idx, n_iter
 ):
-    """
+    r"""
     Maximisation of the pvalue in respect of a nuisance parameter considering
     a 2x2 sample space.
+
+    Barnard and Boschloo exact test iterate over a nuisance parameter \pi
```

Comment:
This is the first occurence of `Boschloo`. If you want to mention this, you have to introduce it.

Reply:
I want to wait that the PR is finished and merge to begin to work on Boschloo, so that I am not changing everything on Boschloo as well. Should I remove Boschloo reference for now then?

Reply:
Yes I would say so. Thanks.